AGILE/Commands.cs

```diff
@@ -1637,9 +1620,9 @@ private int ExecuteAction(Action action, ref byte newRoom, ref bool exit)
                             menu.EnableAllMenus();
                             ReplayScriptEvents();
                             ShowPicture(false);
-                            newRoom = state.CurrentRoom = state.Vars[Defines.CURROOM];
+                            state.CurrentRoom = state.Vars[Defines.CURROOM];
```

Comment:
I think we can now delete all instances of `state.CurrentRoom`

Reply:
Yes, quite possibly. I'll remove it in a separate change.

Reply:
Actually, maybe I will do it with this one, as perhaps the issue needing the CurrentRoom to be added was due to where NewRoom was previously being invoked.

Reply:
Yeah, you're right. The only use of CurrentRoom was in the old NewRoom, and only needed due to it being outside of the main loop. So that was indeed the cause of issue #21 , as you say. Good call.